pointcloud/basic_octree.go

```diff
@@ -129,3 +131,55 @@ func (octree *BasicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Ve
 func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
+
+// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
+func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, threshold int, buffer float64) (bool, error) {
+	if octree.MaxVal() < threshold {
+		return false, nil
+	}
+	switch octree.node.nodeType {
+	case internalNode:
+		ocbox, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(octree.center),
+			r3.Vector{octree.sideLength + buffer, octree.sideLength + buffer, octree.sideLength + buffer},
+			"",
+		)
+		if err != nil {
+			return false, err
+		}
+
+		// Check whether our geom collides with the area represented by the octree. If false, we can skip
+		collide, err := geom.CollidesWith(ocbox)
+		if err != nil {
+			return false, err
+		}
+		if !collide {
+			return false, nil
+		}
+		for _, child := range octree.node.children {
+			collide, err = child.CollidesWithGeometry(geom, threshold, buffer)
+			if err != nil {
+				return false, err
+			}
+			if collide {
+				return true, nil
+			}
+		}
+	case leafNodeEmpty:
+		return false, nil
```

Comment:
[nit] Better to make this the default case I think.  Can move to bottom and drop through to the return at end of file to avoid redundancy

Reply:
In the case where a new node type is added and this function is not updated, would prefer to error rather than simply return "no collision".

Reply:
Does the linter not complain about not having a default though? I think that is the preferred practice. If not I have no issue with that

Either way can still remove the redundant return and fall through

Reply:
The linter complains about not having a default only in the case where the enumerated cases are non-exhaustive.

Prefer not to fallthrough because it's a more complex flow that has to be understood when reading and would become more fragile if code is added for empty nodes. We wouldn't even be saving a line of code.